compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -822,10 +822,6 @@ private boolean isAssignable(BType source, BType target, Set<TypePair> unresolve
             return true;
         }
 
-        if (sourceTag == TypeTags.XML_TEXT && targetTag == TypeTags.CHAR_STRING) {
```

Comment:
Maybe we can move this also to the check above, but then again since this is getting removed anyway, maybe we can leave it here for now?